diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f219e6d..b6ff80ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -454,4 +454,4 @@
 if(MSVC)
-    target_compile_options(operon_operon PRIVATE "/std:c++latest")
-    target_compile_options(operon_core PRIVATE "/std:c++latest")
+    target_compile_options(operon_operon PRIVATE "/bigobj")
+    target_compile_options(operon_core PRIVATE "/bigobj")
 else()
     if (UNIX AND NOT APPLE)
         target_link_options(operon_operon PRIVATE "-Wl,--no-undefined")
